### PR TITLE
DataViews: make dataviews powered page patterns stable

### PR DESCRIPTION
**Pull request by oandregal**
Part of https://github.com/WordPress/gutenberg/issues/55083

## What?

This PR makes the dataviews-powered Patterns page stable.

## Why?

It's considered to have feature-parity with the existing.

## How?

- Removes the check for experiment in the router.
- Updates the contents of `page-patterns/index.js` with those of `page-patterns/dataviews-patterns.js` and removes the latter.

## Testing Instructions

Load the "Side Editor > Patterns" page with and without the "admin views" experiment enabled and verify that the dataviews-powered page is loaded.

Captura de ecrã 2024-01-23, às 18 08 49

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]

File: packages/edit-site/src/components/page-patterns/index.js [hunk 1]
```diff
@@ -1,42 +1,421 @@
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
```
Comment (oandregal) [inline packages/edit-site/src/components/page-patterns/index.js:4]: The changes in this file are just copy/pasting `dataviews-patterns.js`. These can be skipped from the review.

File: packages/edit-site/src/components/layout/router.js [hunk 2]
```diff
@@ -9,9 +9,8 @@ import { privateApis as routerPrivateApis } from '@wordpress/router';
 import { unlock } from '../../lock-unlock';
 import { useIsSiteEditorLoading } from './hooks';
 import Editor from '../editor';
-import DataviewsPatterns from '../page-patterns/dataviews-patterns';
```
Comment (oandregal) [inline packages/edit-site/src/components/layout/router.js:12]: This file is the only that really changes.

File: packages/edit-site/src/components/page-patterns/index.js (hunk 1 above)
Comment (youknowriad) [reply packages/edit-site/src/components/page-patterns/index.js:4]: I was surprised that GitHub didn't say the file moved.

File: test/e2e/specs/site-editor/patterns.spec.js [hunk 3]
```diff
@@ -1,252 +0,0 @@
...
-
-		this.content = this.#page.getByRole( 'region', {
-			name: 'Patterns content',
-		} );
-		this.navigation = this.#page.getByRole( 'region', {
-			name: 'Navigation',
-		} );
-		this.list = this.content.getByRole( 'list' );
-	}
-}
```
Comment (youknowriad) [inline test/e2e/specs/site-editor/patterns.spec.js]: Feels like there's an opportunity to test our new screens better. Maybe we should adapt the tests instead?

File: packages/edit-site/src/components/page-patterns/index.js (hunk 1 above)
Comment (oandregal) [reply packages/edit-site/src/components/page-patterns/index.js:4]: There are changes in other files (imports), not sure if that's the reason.

**Comment by oandregal** [comment]
There are a few things (styles, other components, etc.) that we may want to remove. I'll do that in a separate follow-up, to make sure I don't miss any styles.

File: test/e2e/specs/site-editor/patterns.spec.js (hunk 3 above)
Comment (oandregal) [reply test/e2e/specs/site-editor/patterns.spec.js]: Yeah, we need to add tests for every new page. I am unsure if I'll have time for that before RC, but can commit to do it in a follow-up.

Comment (ntsekouras) [reply test/e2e/specs/site-editor/patterns.spec.js]: I agree we should adapt the tests. If the plan is to do it a follow up, maybe just skip them for now?

**Comment by talldan** [comment]
Gave it a quick test. Didn't spot any issues, and I really enjoyed the improved UX 👍

File: test/e2e/specs/site-editor/patterns.spec.js (hunk 3 above)
Comment (oandregal) [reply test/e2e/specs/site-editor/patterns.spec.js]: Sure, I can skip them https://github.com/WordPress/gutenberg/pull/58139/commits/4506147debca570cd989b6a13466e126ea1fb205 Though I'm skeptical we can reuse any of that code (that's why I removed in the first place).

Comment (oandregal) [reply test/e2e/specs/site-editor/patterns.spec.js]: I added tasks for adding e2e test for every page in the tracking issue.

**Comment by oandregal** [comment]
Milestoned this for 17.6, so it's not missed.

**Pull request merged**